timer/src/main/java/com/toggl/timer/log/domain/TimeEntriesLogReducer.kt

```diff
@@ -102,6 +102,11 @@ class TimeEntriesLogReducer @Inject constructor(private val repository: TimeEntr
                     state.value = state.value.copy(expandedGroupIds = newUngroupedTimeEntries)
                     noEffect()
                 }
+                is TimeEntriesLogAction.UndoButtonPressed -> {
+                    if (state.value.entriesPendingDeletion.isNotEmpty())
+                        state.value = state.value.copy(entriesPendingDeletion = setOf())
```

Comment:
Why do we even need this check?
Just set it to `emptySet` and call it a day :)
```suggestion
                    state.value = state.value.copy(entriesPendingDeletion = emptySet())
```

Reply:
I wasn't sure how it worked here, coming from react world. I just wanted to make sure we don't needlessly update the state with updates that don't do anything, since that might trigger some subscribers to run computations and stuff. 

For instance (and yeah, I should maybe just test it) - if I reset it to `emptySet`, when it already is empty, will the time entries selector rerun? If it does, then I'd say we leave the check in place.